keep/providers/splunk_provider/splunk_provider.py

```diff
@@ -0,0 +1,154 @@
+import dataclasses
+
+import pydantic
+from splunklib.client import connect
+
+from keep.api.models.alert import AlertDto, AlertSeverity
+from keep.contextmanager.contextmanager import ContextManager
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig, ProviderScope
+from keep.providers.providers_factory import ProvidersFactory
+
+
+@pydantic.dataclasses.dataclass
+class SplunkProviderAuthConfig:
+    username: str = dataclasses.field(
+        metadata={
+            "required": True,
+            "description": "Splunk Username",
+        }
+    )
+    password: str = dataclasses.field(
+        metadata={
+            "required": True,
+            "description": "Splunk Password",
+            "sensitive": True,
+        }
+    )
+
+    host: str = dataclasses.field(
+        metadata={
+            "description": "Splunk Host (default is localhost)",
+        },
+        default="localhost"
+    )
+    port: int = dataclasses.field(
+        metadata={
+            "description": "Splunkd Port (default is 8089)",
+        },
+        default=8089
+    )
+
+
+class SplunkProvider(BaseProvider):
+    """Pull alerts and query incidents from Splunk."""
+
+    PROVIDER_SCOPES = [
+        ProviderScope(
+            name="connect_to_client",
+            description="The user can connect to the client",
+            mandatory=True,
+            alias="Connect to the client",
+        )
```

Comment:
Can we also determine if:
1. The user can see alert definitions
2. The user can edit alert definitions
This will make more sense then just connecting..

Reply:
Yup we can do that. Permissions needed (Bare min) `list_all_objects` and `edit_own_objects` (guess)